### DISCUSSION TITLE
Get JSON field only if it exists

### DISCUSSION THREAD

**Discussion by RobinFrcd** [ClickHouse/ClickHouse#70057] [Q&A]
Hi,
I have a table with a `Object('json')` column. I would like to get the values for given keys if they exist, and `null` if not. I tried `if(has(tupleNames(data), 'test'), data.test, NULL)` but the true value (`data.test`) gets evaluated even if data.test does not exist:

```
Code: 47. DB::Exception: Missing columns: 'data.test' while processing query: 'SELECT if(has(tupleNames(data), 'test'), data.test, NULL) FROM bi.events', required columns: 'data' 'data.test', maybe you meant: 'data'. (UNKNOWN_IDENTIFIER) (version 24.9.1.3278 (official build))
```

Is there a simple way to get a JSON value only if it exists for a given row ? Thanks !

**Comment by matt-best-elateral** [comment]
Does `isValidJSON` work in your scenario?

https://clickhouse.com/docs/en/sql-reference/functions/json-functions#isvalidjson
